src/v/cluster/metadata_dissemination_service.cc

```diff
@@ -384,26 +384,10 @@ ss::future<> metadata_dissemination_service::dispatch_disseminate_leadership() {
 ss::future<> metadata_dissemination_service::update_leaders_with_health_report(
   cluster_health_report report) {
     vlog(clusterlog.trace, "updating leadership from health report");
-    for (const auto& node_report : report.node_reports) {
+    for (auto& node_report : report.node_reports) {
         co_await _leaders.invoke_on_all(
-          [&node_report](partition_leaders_table& leaders) {
-              for (auto& tp : node_report.topics) {
-                  for (auto& p : tp.partitions) {
-                      // Nodes may report a null leader if they're out of
-                      // touch, even if the leader is actually still up.  Only
-                      // trust leadership updates from health reports if
-                      // they're non-null (non-null updates are safe to apply
-                      // in any order because update_partition leader will
-                      // ignore old terms)
-                      if (p.leader_id.has_value()) {
-                          leaders.update_partition_leader(
-                            model::ntp(tp.tp_ns.ns, tp.tp_ns.tp, p.id),
-                            p.revision_id,
-                            p.term,
-                            p.leader_id);
-                      }
-                  }
-              }
+          [report = std::move(node_report)](partition_leaders_table& leaders) {
```

Comment:
FYI I believe invoke_on_all uses `std::function` and captures may be copied. So if you're trying to prevent copies you probably want to keep the reference capture?

Reply:
i will change that